$NetBSD: patch-configure,v 1.2 2020/08/30 19:26:45 christos Exp $

* Add DragonFly support.
* Use proper link options for NetBSD.
* Link proper postgresql library.
* Take from NetBSD base, adding support for blacklistd.

--- configure.orig	2020-07-03 06:44:14.000000000 -0400
+++ configure	2020-08-30 15:10:37.131653135 -0400
@@ -637,6 +637,10 @@
 BUILD_CPPFLAGS
 BUILD_CFLAGS
 BUILD_CC
+BLACKLISTLINKOBJS
+BLACKLISTLINKSRCS
+BLOCKLISTLINKOBJS
+BLOCKLISTLINKSRCS
 DLZ_DRIVER_MYSQL_LIBS
 DLZ_DRIVER_MYSQL_INCLUDES
 DLZ_SYSTEM_TEST
@@ -732,6 +736,8 @@
 JSON_C_CFLAGS
 LIBXML2_LIBS
 LIBXML2_CFLAGS
+LIBUV_LIBS
+LIBUV_CFLAGS
 NZD_MANS
 NZDTARGETS
 NZDSRCS
@@ -1719,6 +1725,8 @@
   --with-dlz-stub=ARG     Build with stub DLZ driver [yes|no]. (Required to
                           use stub driver with DLZ)
   --with-make-clean       run "make clean" at end of configure [yes|no]
+  --with-blacklist	  Build with blacklist
+  --with-blocklist	  Build with blocklist
 
 Some influential environment variables:
   CC          C compiler command
@@ -3917,7 +3925,7 @@
 
 # On IRIX 5.3, sys/types and inttypes.h are conflicting.
 for ac_header in sys/types.h sys/stat.h stdlib.h string.h memory.h strings.h \
-		  inttypes.h stdint.h unistd.h
+		  inttypes.h stdint.h unistd.h blacklist.h blocklist.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_compile "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default
@@ -18385,7 +18393,265 @@
 
 
 
+#
+# was --with-blacklist specified?
+#
+BLACKLISTLINKOBJS=
+BLACKLISTLINKSRCS=
+
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for blacklist library" >&5
+$as_echo_n "checking for blacklist library... " >&6; }
+
+# Check whether --with-blacklist was given.
+if test "${with_blacklist+set}" = set; then :
+  withval=$with_blacklist; with_blacklist="$withval"
+else
+  with_blacklist="auto"
+fi
+
+
+have_blacklist=""
+case "$with_blacklist" in
+	no)
+		;;
+	auto|yes)
+		for d in /usr /usr/local /opt/local
+		do
+			if test -f "${d}/include/blacklist.h"
+			then
+				if test ${d} != /usr
+				then
+					blacklist_cflags="-I ${d}/include"
+					LIBS="$LIBS -L${d}/lib"
+				fi
+				have_blacklist="yes"
+			fi
+		done
+		;;
+	*)
+		if test -f "${with_blacklist}/include/blacklist.h"
+		then
+			blacklist_cflags="-I${with_blacklist}/include"
+			LIBS="$LIBS -L${with_blacklist}/lib"
+			have_blacklist="yes"
+		else
+			as_fn_error $? "$with_blacklist/include/blacklist.h not found." "$LINENO" 5
+		fi
+		;;
+esac
+
+if test "X${have_blacklist}" != "X"
+then
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
+$as_echo "yes" >&6; }
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for library containing blacklist" >&5
+$as_echo_n "checking for library containing blacklist... " >&6; }
+if ${ac_cv_search_blacklist+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  ac_func_search_save_LIBS=$LIBS
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
+#ifdef __cplusplus
+extern "C"
+#endif
+char blacklist_open(void);
+int
+main ()
+{
+return blacklist_open();
+  ;
+  return 0;
+}
+_ACEOF
+for ac_lib in '' blacklist; do
+  if test -z "$ac_lib"; then
+    ac_res="none required"
+  else
+    ac_res=-l$ac_lib
+    LIBS="-l$ac_lib  $ac_func_search_save_LIBS"
+  fi
+  if ac_fn_c_try_link "$LINENO"; then :
+  ac_cv_search_blacklist=$ac_res
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext
+  if ${ac_cv_search_blacklist+:} false; then :
+  break
+fi
+done
+if ${ac_cv_search_blacklist+:} false; then :
+
+else
+  ac_cv_search_blacklist=no
+fi
+rm conftest.$ac_ext
+LIBS=$ac_func_search_save_LIBS
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_search_blacklist" >&5
+$as_echo "$ac_cv_search_blacklist" >&6; }
+ac_res=$ac_cv_search_blacklist
+if test "$ac_res" != no; then :
+  test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
+
+else
+  as_fn_error $? "found blacklist include but not library." "$LINENO" 5
+			have_blacklist=""
+fi
+
+elif test "X$with_blacklist" = Xyes
+then
+	as_fn_error $? "include/blacklist.h not found." "$LINENO" 5
+else
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
+$as_echo "no" >&6; }
+fi
+
+BLACKLIST=
+if test "X${have_blacklist}" != "X"
+then
+	CFLAGS="$CFLAGS $blacklist_cflags"
+
+$as_echo "#define HAVE_BLACKLIST 1" >>confdefs.h
+
+	BLACKLIST=1
+	BLACKLISTLINKSRCS='${BLACKLISTLINKSRCS}'
+	BLACKLISTLINKOBJS='${BLACKLISTLINKOBJS}'
+fi
+
+#
+# was --with-blocklist specified?
+#
+BLOCKLISTLINKOBJS=
+BLOCKLISTLINKSRCS=
+
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for blocklist library" >&5
+$as_echo_n "checking for blocklist library... " >&6; }
+
+# Check whether --with-blocklist was given.
+if test "${with_blocklist+set}" = set; then :
+  withval=$with_blocklist; with_blocklist="$withval"
+else
+  with_blocklist="auto"
+fi
+
+
+have_blocklist=""
+case "$with_blocklist" in
+	no)
+		;;
+	auto|yes)
+		for d in /usr /usr/local /opt/local
+		do
+			if test -f "${d}/include/blocklist.h"
+			then
+				if test ${d} != /usr
+				then
+					blocklist_cflags="-I ${d}/include"
+					LIBS="$LIBS -L${d}/lib"
+				fi
+				have_blocklist="yes"
+			fi
+		done
+		;;
+	*)
+		if test -f "${with_blocklist}/include/blocklist.h"
+		then
+			blocklist_cflags="-I${with_blocklist}/include"
+			LIBS="$LIBS -L${with_blocklist}/lib"
+			have_blocklist="yes"
+		else
+			as_fn_error $? "$with_blocklist/include/blocklist.h not found." "$LINENO" 5
+		fi
+		;;
+esac
+
+if test "X${have_blocklist}" != "X"
+then
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
+$as_echo "yes" >&6; }
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for library containing blocklist" >&5
+$as_echo_n "checking for library containing blocklist... " >&6; }
+if ${ac_cv_search_blocklist+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  ac_func_search_save_LIBS=$LIBS
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
+#ifdef __cplusplus
+extern "C"
+#endif
+char blocklist_open(void);
+int
+main ()
+{
+return blocklist_open();
+  ;
+  return 0;
+}
+_ACEOF
+for ac_lib in '' blocklist; do
+  if test -z "$ac_lib"; then
+    ac_res="none required"
+  else
+    ac_res=-l$ac_lib
+    LIBS="-l$ac_lib  $ac_func_search_save_LIBS"
+  fi
+  if ac_fn_c_try_link "$LINENO"; then :
+  ac_cv_search_blocklist=$ac_res
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext
+  if ${ac_cv_search_blocklist+:} false; then :
+  break
+fi
+done
+if ${ac_cv_search_blocklist+:} false; then :
+
+else
+  ac_cv_search_blocklist=no
+fi
+rm conftest.$ac_ext
+LIBS=$ac_func_search_save_LIBS
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_search_blocklist" >&5
+$as_echo "$ac_cv_search_blocklist" >&6; }
+ac_res=$ac_cv_search_blocklist
+if test "$ac_res" != no; then :
+  test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
+
+else
+  as_fn_error $? "found blocklist include but not library." "$LINENO" 5
+			have_blocklist=""
+fi
+
+elif test "X$with_blocklist" = Xyes
+then
+	as_fn_error $? "include/blocklist.h not found." "$LINENO" 5
+else
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
+$as_echo "no" >&6; }
+fi
+
+BLOCKLIST=
+if test "X${have_blocklist}" != "X"
+then
+	CFLAGS="$CFLAGS $blocklist_cflags"
+
+$as_echo "#define HAVE_BLOCKLIST 1" >>confdefs.h
 
+	BLOCKLIST=1
+	BLOCKLISTLINKSRCS='${BLOCKLISTLINKSRCS}'
+	BLOCKLISTLINKOBJS='${BLOCKLISTLINKOBJS}'
+fi
 # Check whether --with-zlib was given.
 if test "${with_zlib+set}" = set; then :
   withval=$with_zlib;
@@ -22517,7 +22783,7 @@
 
 fi
 	        ;; #(
-  *-freebsd*|*-openbsd*|*-netbsd*) :
+  *-dragonfly*|*-freebsd*|*-openbsd*|*-netbsd*) :
 
 		 LDFLAGS="${LDFLAGS} -Wl,-E"
 		 SO_CFLAGS="-fpic"
@@ -22549,9 +22815,9 @@
 	        ;; #(
   *-solaris*) :
 
-		 SO_CFLAGS="-KPIC"
-		 SO_LDFLAGS="-G -z text"
-		 SO_LD="ld"
+		 SO_CFLAGS="-fPIC"
+		 SO_LDFLAGS="-Xcompiler -shared -Wl,-z -Wl,text"
+		 SO_LD="${CC}"
 	        ;; #(
   ia64-hp-hpux*) :
 
@@ -22980,8 +23246,8 @@
 	fi
 	if test -n "-L$use_dlz_postgres_lib -lpq"
 	then
-		DLZ_DRIVER_LIBS="$DLZ_DRIVER_LIBS -L$use_dlz_postgres_lib -lpq"
-		DLZ_DRIVER_POSTGRES_LIBS="-L$use_dlz_postgres_lib -lpq"
+		DLZ_DRIVER_LIBS="$DLZ_DRIVER_LIBS -L${PREFIX}/lib -lpq"
+		DLZ_DRIVER_POSTGRES_LIBS="-L${PREFIX}/lib -lpq"
 	fi
 
 
@@ -24465,6 +24731,8 @@
   case $ac_option in
   # Handling of the options.
   -recheck | --recheck | --rechec | --reche | --rech | --rec | --re | --r)
+	: Avoid regenerating within pkgsrc
+	exit 0
     ac_cs_recheck=: ;;
   --version | --versio | --versi | --vers | --ver | --ve | --v | -V )
     $as_echo "$ac_cs_version"; exit ;;
@@ -26186,6 +26454,8 @@
 	test "X$PYTHON" = "X" || echo "    Python tools (--with-python)"
 	test "X$LIBXML2_LIBS" = "X" || echo "    XML statistics (--with-libxml2)"
 	test "X$JSON_C_LIBS" = "X" || echo "    JSON statistics (--with-json-c): $JSON_C_CFLAGS $JSON_C_LIBS"
+	test "X$BLACKLIST" = "X" || echo "    blacklist support (--with-blacklist)"
+	test "X$BLOCKLIST" = "X" || echo "    blocklist support (--with-blocklist)"
 	test "X$ZLIB_LIBS" = "X" || echo "    HTTP zlib compression (--with-zlib)"
 	test "X$NZD_TOOLS" = "X" || echo "    LMDB database to store configuration for 'addzone' zones (--with-lmdb)"
 	test "no" = "$with_libidn2" || echo "    IDN support (--with-libidn2)"
@@ -26264,6 +26534,8 @@
     test "X$PYTHON" = "X" && echo "    Python tools (--with-python)"
     test "X$LIBXML2_LIBS" = "X" && echo "    XML statistics (--with-libxml2)"
     test "X$JSON_C_LIBS" = "X" && echo "    JSON statistics (--with-json-c)"
+    test "X$BLACKLIST" = "X" && echo "    blacklist support (--with-blacklist)"
+    test "X$BLOCKLIST" = "X" && echo "    blocklist support (--with-blocklist)"
     test "X$ZLIB_LIBS" = "X" && echo "    HTTP zlib compression (--with-zlib)"
     test "X$NZD_TOOLS" = "X" && echo "    LMDB database to store configuration for 'addzone' zones (--with-lmdb)"
     test "no" = "$with_libidn2" && echo "    IDN support (--with-libidn2)"
